initialization/global_handlers.py

```diff
@@ -0,0 +1,54 @@
+import datetime
+from uuid import uuid4
+
+from flask import g, session
+
+from db import get_connection, get_session
+
+
+def init_global_handlers(app):
+    app.before_request(_make_session_permanent)
+    app.before_request(_open_db_connection)
+    app.before_request(_fetch_user)
+
+    app.after_request(_close_db_connection)
+
+    return app
+
+
+def _make_session_permanent():
+    # By default, expires in 31 days
+    session.permanent = True
+
+
+def _open_db_connection():
+    if 'db_conn' not in g:
+        g.db_conn    = get_connection()
+        g.db_session = get_session(g.db_conn)
+
+
+def _fetch_user():
+    if 'user' not in g:
+        if 'user_uuid' in session:
+            user_uuid = session['user_uuid']
+        else:
+            # Create a new user UUID so we can keep track of this browser
+            # TODO might be temporary, might be useful to convert anonymous users
+            user_uuid = str(uuid4())
+
+        session['user_uuid'] = user_uuid
+        g.current_user = User(uuid=user_uuid)
```

Comment:
Eventually, the "current user" is going to be stored in the database with the user id stored in the (encrypted) session cookie. So, this scheme should work fine even when we have "real" users, we simply won't generate a random UUID, but we'll store the user id after login.

Or, we could always generate a random UUID and the login action would simply persist it to the database. That way, an anonymous user will still have an ID and submission persistence for some time. We can decide how to tackle this later.

Reply:
> anonymous user will still have an ID and submission persistence for some time

Maybe avoid this. Since we go with users, better to always ask them to make an account first

Reply:
Maybe you're right, and if we use ORCID, it will be quite easy. Let's see how it goes when I get around to implementing the full user registration flow.